### PR TITLE
Allow verbose output of structure of test and provide a simple test reducer

### PR DESCRIPTION
**Pull request by agroce**
This allows DeepState to output information on the structure (when bytes are read, when OneOf calls start and finish) of a test execution, for debugging and use by an external test case reducer (see #109).  In the long run, we'll want a version that provides similar information in a structure that can be populated during execution, for use by "inline" reducers like custom libFuzzer mutators.

### CODE REVIEW

**Comment by agroce** [comment]
Now includes a simple test case reducer for deepstate, based on OneOfs and byte lowering.  Should also include byte removal, as future work, but I expect that to usually fail.

File: bin/deepstate/reducer.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import sys
+import subprocess
+
```
Comment (pgoodman) [inline bin/deepstate/reducer.py]: Lets make this use `argparse`, and have a copyright notice. Ideally also two-space indent to be consistent with the rest of the python code.

File: src/lib/DeepState.c [hunk 2]
```diff
@@ -140,6 +141,9 @@ void DeepState_SymbolizeData(void *begin, void *end) {
       if (DeepState_InputIndex >= DeepState_InputSize) {
         DeepState_Abandon("Read too many symbols");
       }
+      if (FLAGS_verbose_reads) {
+	printf ("Reading byte at %u\n", DeepState_InputIndex);
```
Comment (pgoodman) [inline src/lib/DeepState.c]: Indent the `printf` using spaces, not tabs

Comment (pgoodman) [reply src/lib/DeepState.c]: No space between `printf` and `(`, i.e. just have `printf(`

File: src/lib/DeepState.c [hunk 3]
```diff
@@ -227,6 +234,9 @@ int DeepState_Bool(void) {
       type val = 0; \
       _Pragma("unroll") \
       for (size_t i = 0; i < sizeof(type); ++i) { \
+          if (FLAGS_verbose_reads) { \
```
Comment (pgoodman) [inline src/lib/DeepState.c]: Outdent the `if` statement to be in line with `val = ...`

**Comment by agroce** [comment]
Will fix these up as I get a chance.

**Comment by agroce** [comment]
(argparse and the indent, that is)

**Comment by agroce** [comment]
On indentation, I was thinking we could go autopep8 and use pycodestyle too, possibly.

**Comment by agroce** [comment]
Gah, there doesn't seem to be a nice way to change the indentation other than manually.  If I change emacs py-indent-offset, then use indent-region or py-indent-region, for some reason it totally hoses the code.  I'll do this later, if we decide not to go pep8.

**Comment by agroce** [comment]
This should be ready.  The reducer could be much smarter, but it's decently effective now, and being much more effective probably means things like noticing patterns (e.g. if you mkdir "aaa" in one spot and rmdir "aaa" in another spot, you need to reduce those "aaa"s at the same time).

**Pull request merged**